tasks/kmt.py

```diff
@@ -269,55 +272,141 @@ def full_arch(arch):
     return arch
 
 
+def build_tests_package(ctx, source_dir, stack, arch, ci, verbose=True):
+    root = os.path.join(source_dir, "kmt-deps")
+    test_archive = f"tests-{arch}.tar.gz"
+    if not ci:
+        system_probe_tests = os.path.join(root, stack, "opt/system-probe-tests")
+        test_pkgs = os.path.join(
+            source_dir, "test/kitchen/site-cookbooks/dd-system-probe-check/files/default/tests/pkg"
+        )
+        ctx.run(f"rm -rf {system_probe_tests} && mkdir -p {system_probe_tests}", hide=(not verbose))
+        ctx.run(f"cp -R {test_pkgs} {system_probe_tests}", hide=(not verbose))
+        with ctx.cd(os.path.join(root, stack)):
+            ctx.run(f"tar czvf {test_archive} opt", hide=(not verbose))
+
+
 @task
-def prepare(ctx, vms, stack=None, arch=None, ssh_key=None, rebuild_deps=False, packages="", verbose=True):
+def build_dependencies(ctx, arch, layout_file, source_dir, ci=False, stack=None, verbose=True):
+    root = os.path.join(source_dir, "kmt-deps")
+    deps_dir = os.path.join(root, "dependencies")
+    if not ci:
+        if stack is None:
+            raise Exit("no stack name provided")
+        deps_dir = os.path.join(root, stack, "dependencies")
+        # in the CI we can rely on gotestsum being present
+        download_gotestsum(ctx)
+
+    if os.path.exists(deps_dir):
+        shutil.rmtree(deps_dir)
+
+    Path(deps_dir).mkdir(parents=True)
+
+    with open(layout_file) as f:
+        deps_layout = json.load(f)
+    with ctx.cd(deps_dir):
+        for new_dirs in deps_layout["layout"]:
+            ctx.run(f"mkdir -p {new_dirs}", hide=(not verbose))
```

Comment:
nit: here you use `ctx.run("mkdir")` whereas you use `Path.mkdir()` line 303. I don't know which one we should prefer, maybe we want to be uniform

Reply:
Opting for `ctx` since this is the one used normally.